src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,148 @@
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractInternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectGuidKey = nameof(ProjectGuidKey);
+
+        protected abstract bool IsPositionEntirelyWithinStringLiteral(SyntaxTree syntaxTree, int position, CancellationToken cancellationToken);
+        protected abstract string GetAttributeNameOfAttributeSyntaxNode(SyntaxNode attributeSyntaxNode);
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var syntaxTree = await context.Document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFactsService = context.Document.GetLanguageService<ISyntaxFactsService>();
+            if (IsPositionEntirelyWithinStringLiteral(syntaxTree, context.Position, cancellationToken))
+            {
+                var token = syntaxTree.FindTokenOnLeftOfPosition(context.Position, cancellationToken);
+                var attributeSyntaxNode = GetAttributeSyntaxNodeOfToken(syntaxFactsService, token);
+                if (attributeSyntaxNode == null)
+                {
+                    return;
+                }
+
+                var attributeName = GetAttributeNameOfAttributeSyntaxNode(attributeSyntaxNode);
+                if (attributeName == "InternalsVisibleTo" || attributeName == nameof(InternalsVisibleToAttribute))
+                {
+                    if (await CheckTypeInfoOfAttributeAsync(context, attributeSyntaxNode).ConfigureAwait(false))
+                    {
+                        AddAssemblyCompletionItems(context, cancellationToken);
+                    }
+                }
+            }
+        }
+
+        private static SyntaxNode GetAttributeSyntaxNodeOfToken(ISyntaxFactsService syntaxFactsService, SyntaxToken token)
+        {
+            //Supported cases:
+            //[Attribute("|
+            //[Attribute(parameterName:"Text|")
+            //Also supported but excluded by syntaxTree.IsEntirelyWithinStringLiteral in ProvideCompletionsAsync
+            //[Attribute(""|
+            //[Attribute("Text"|)
+            var node = token.Parent;
+            if (syntaxFactsService.IsStringLiteralExpression(node))
+            {
+                // LiteralExpressionSyntax -> AttributeArgumentSyntax -> AttributeArgumentListSyntax -> AttributeSyntax
+                var attributeSyntaxNodeCandidate = node.Parent?.Parent?.Parent;
+                if (attributeSyntaxNodeCandidate != null && syntaxFactsService.IsAttribute(attributeSyntaxNodeCandidate))
```

Comment:
syntaxFacts methods should be resilient to null for their IsXXX methods, so no need for null check here. (if the method is not resilient, then make it resilient).